fss/src/app/children/child-details.component.ts

```diff
@@ -93,16 +94,33 @@ export class ChildDetails implements OnInit, OnChanges {
         this.restService.getEnum("fss_child_status").then(status => this.child_status = status);
         this.restService.changeEmitter.subscribe(() => this.ngOnChanges());
         this.child_photo_url = this.restService.getChildPhotoUrl(this.child_id);
+        setInterval(()=>this.autosave(), this.restService.autosave_frequency);
     }
     ngOnChanges(): void {
         this.restService.getChild(this.child_id).then(child => {
             if (child == undefined) return;
             this.child = child;
+            this.orig_child = Object.assign(Object.create(child), child);
             this.age = this.getAgeStr()
             this.child.birth_date_object = this.restService.getDateFromString(this.child.birth_date)
         });
         this.child_photo_url = this.restService.getChildPhotoUrl(this.child_id) + "?" + this.on_changes_count++;
     }
+    autosave(): void {
```

Comment:
Not a bad idea - but I would move the equality comparison into the Child class. That way you can say: 
`this.child.equals(this.orig_child)` in here.